pagerduty/user.go

```diff
@@ -383,14 +408,42 @@ func (s *UserService) DeleteContactMethod(userID, contactMethodID string) (*Resp
 	return resp, err
 }
 
+// ListContactMethods lists contact methods for a user.
+func (s *UserService) ListNotificationRules(userID string) (*ListNotificationRulesResponse, *Response, error) {
+	u := fmt.Sprintf("/users/%s/notification_rules", userID)
+	v := new(ListNotificationRulesResponse)
+
+	resp, err := s.client.newRequestDo("GET", u, nil, nil, &v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
 // CreateNotificationRule creates a new notification rule for a user.
 func (s *UserService) CreateNotificationRule(userID string, rule *NotificationRule) (*NotificationRule, *Response, error) {
 	u := fmt.Sprintf("/users/%s/notification_rules", userID)
 	v := new(NotificationRulePayload)
 
 	resp, err := s.client.newRequestDo("POST", u, nil, &NotificationRulePayload{NotificationRule: rule}, &v)
 	if err != nil {
-		return nil, nil, err
+		if e, ok := err.(*Error); ok && strings.Compare(fmt.Sprintf("%v", e.Errors), "[Channel Start delay must be unique for a given contact method]") == 0 {
+			resp, _, lErr := s.ListNotificationRules(userID)
+			if lErr != nil {
+				return nil, nil, fmt.Errorf("user notification rule is not unique and failed to fetch existing ones: %w", lErr)
+			}
+
+			for _, notification := range resp.NotificationRules {
```

Comment:
It's a rule, not a notification, maybe nr in the style of golang?
```suggestion
			for _, nr := range resp.NotificationRules {
```

Reply:
Good catch, thanks!